src/rpc/blockchain.cpp

```diff
@@ -1423,6 +1423,77 @@ UniValue z_gettreestate(const UniValue& params, bool fHelp)
     return res;
 }
 
+UniValue z_getsubtreesbyindex(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() < 2 || params.size() > 3)
+        throw runtime_error(
+            "z_getsubtreesbyindex \"pool\" start_index ( end_index )\n"
+            "Return information about the given block's tree state.\n"
+            "\nArguments:\n"
+            "1. \"pool\"        (string, required) The pool from which subtrees should be returned. Either \"sapling\" or \"orchard\".\n"
+            "2. start_index   (numeric, required) The index of the first subtree to return.\n"
+            "2. end_index     (numeric, optional) The index of the last subtree to return.\n"
```

Comment:
Currently `end_index` is an inclusive upper bound, as that seemed more natural for it being an explicit JSON-RPC parameter. I'm open to changing it to an exclusive upper bound if that would be more comprehensible or preferred.

Reply:
Maybe the parameters should be `start_index` and `count` instead; I think that might be more natural.

Reply:
I'm fine with either an exclusive upper bound or a count. I dislike inclusive upper bounds (see https://www.cs.utexas.edu/users/EWD/transcriptions/EWD08xx/EWD831.html).